### ISSUE TITLE
To ignore everything in footer

### ISSUE THREAD

**Issue by chantorak** [html-to-text/node-html-to-text#295]
**The goal**
*To ignore nested p in footer*

**Best attempt**
This is my selector
```
{
  baseElements: {
     selectors: ["p"],
  },
  selectors: [
    {selector: "p", format: "paragraph"},
    {selector: "footer > *", format: "skip"}
  ]
}
```

**The question**
```
<footer>
   <div>
      <p>Copyright text 1</p>
      <div>
         <p>Copyright text 2</p>
      </div>
   </div>
</footer>
```
Is there a way of targeting everything on a footer to get ignored?
instead of having to define multiple skips for each level

**Comment by KillyMXI** [comment]
Selectors in `html-to-text` is not as advanced as you might expect in browsers.

` ` (space) combinator is not supported currently - it can be quite a bit more computationally expensive than `>` and cause slowdowns when misused, unless/until a more comprehensive system is implemented.

It would've been easy to skip the `footer` tag. But you're selecting base elements and that happens in a preceding step. Not sure if I want to change this behavior.
It might become possible to restrict base element selectors later with `:not()` when I implement pseudo-classes.

Currently, your best bet is
```jsonc
  {
    baseElements: { selectors: ['p'], },
    selectors: [
      { selector: 'p', format: 'paragraph' },
      { selector: 'footer > p', format: 'skip' },
      { selector: 'footer > * > p', format: 'skip' },
      { selector: 'footer > * > * > p', format: 'skip' },
      { selector: 'footer > * > * > * > p', format: 'skip' }
      // etc
    ]
  }
```

**Comment by KillyMXI** [comment]
Oh yeah, that's clever 👍

**Comment by KillyMXI** [comment]
I'm a bit jealous I didn't come with the same solution myself.
At the same time, it feels great to know I designed a system powerful enough I don't always see all the possibilities myself.

The solution is more "clever" than "logical", but the fact it works is not a coincidence but the result of how each system is designed to work, so it is reliable.

Closing this now, since there are no actionable items left for me here.
